src/library/tactic/dsimplify.cpp

```diff
@@ -343,11 +343,14 @@ optional<pair<expr, bool>> dsimplify_fn::post(expr const & e) {
     while (true) {
         check_system("dsimplify");
         inc_num_steps();
-        list<simp_lemma> const * simp_lemmas_ptr = m_simp_lemmas.find(curr_e);
-        if (!simp_lemmas_ptr) break;
+        // give the `eq` lemmas priority over the `iff`s, to match what `simp` does.
         buffer<simp_lemma> simp_lemmas;
-        to_buffer(*simp_lemmas_ptr, simp_lemmas);
-
+        if (list<simp_lemma> const * simp_lemmas_ptr = m_simp_lemmas.find(get_eq_name(), curr_e)) {
+            to_buffer(*simp_lemmas_ptr, simp_lemmas);
+        }
+        if (list<simp_lemma> const * simp_lemmas_ptr = m_simp_lemmas.find(get_iff_name(), curr_e)) {
+            to_buffer(*simp_lemmas_ptr, simp_lemmas);
+        }
```

Comment:
You could still include the early break (`if (!simp_lemmas_ptr) break;`) in the form of
```c++
if (simp_lemmas.empty()) break;
```
though this is a micro-optimization since `new_e == curr_e` does start with a pointer check after just a couple layers of indirection.

Reply:
Yeah, I concluded this was a micro-optimization so there was no point.